src/Transport/Topology/TopologyOperator.cs

```diff
@@ -5,15 +5,21 @@ namespace NServiceBus.Transport.AzureServiceBus
     using System.Collections.Generic;
     using System.Threading.Tasks;
     using Logging;
+    using Microsoft.ServiceBus.Messaging;
     using Settings;
 
     class TopologyOperator : IOperateTopologyInternal, IDisposable
     {
         public TopologyOperator(MessageReceiverLifeCycleManager clientEntities, BrokeredMessagesToIncomingMessagesConverter brokeredMessageConverter, ReadOnlySettings settings)
         {
-            readOnlySettings = settings;
             this.brokeredMessageConverter = brokeredMessageConverter;
             messageReceiverLifeCycle = clientEntities;
+
+            messageReceiverNotifierSettings = new MessageReceiverNotifierSettings(
+                settings.Get<ReceiveMode>(WellKnownConfigurationKeys.Connectivity.MessageReceivers.ReceiveMode),
+                settings.HasExplicitValue<TransportTransactionMode>() ? settings.Get<TransportTransactionMode>() : settings.SupportedTransactionMode(),
+                settings.Get<TimeSpan>(WellKnownConfigurationKeys.Connectivity.MessageReceivers.AutoRenewTimeout),
+                settings.Get<int>(WellKnownConfigurationKeys.Connectivity.NumberOfClientsPerEntity));
```

Comment:
is it guaranteed that settings are complete when this constructor is called?

Reply:
I've confirmed that the settings defined during configuration time for an endpoint are the settings passed into TopologyOperator and MessageRecieveNotifier later @yvesgoeleven. Here's the chain of calls:

`MessagePump.Start()` -> `TopologyOperator.Start()` -> `TopologyOperator.StartNotifiers()` -> `TopologyOperator.CreateNotifier()` -> `MessageReceiverNotifier.ctor(notifierSettings)`

`MessagePump.Init()` -> `MessagePump.DetermineTopologyOperator()` -> `TopologyOperator.ctor(settings)`

Settings passed into `MessagePump` ctor are already `ReadOnlySettings`, indicating that configuration stage is finished.

I.e. an endpoint will not be running and using settings until  endpoint configuration is finalized and endpoint is started (message pump is initialized and started).